qutebrowser/browser/greasemonkey.py

```diff
@@ -0,0 +1,289 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016-2017 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Load, parse and make avalaible greasemonkey scripts."""
+
+import re
+import os
+import json
+from fnmatch import fnmatch
+from functools import partial
```

Comment:
Please only import modules, not things from modules (i.e. use `fnmatch.fnmatch` and `functools.partial`)

Reply:
Okay. But why?

Reply:
Readability - if I see `partial(...)` in a file, I expect to find `partial` defined inside that file.

Reply:
Technically, `from fnmatch import fnmatch` _is_ a definition :wink: